homeassistant/components/ps4/config_flow.py

```diff
@@ -0,0 +1,124 @@
+"""Config Flow for PlayStation 4."""
+from collections import OrderedDict
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.ps4.const import (
+    DEFAULT_NAME, DEFAULT_REGION, DOMAIN, REGIONS)
+from homeassistant.const import (
+    CONF_CODE, CONF_HOST, CONF_IP_ADDRESS, CONF_NAME, CONF_REGION, CONF_TOKEN)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class PlayStation4FlowHandler(config_entries.ConfigFlow):
+    """Handle a PlayStation 4 config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.creds = None
+        self.name = None
+        self.host = None
+        self.region = None
+        self.pin = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a user config flow."""
+        from pyps4_homeassistant import Helper
+        helper = Helper()
+
+        # Abort if device is configured.
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='devices_configured')
+
+        # Check if able to bind to ports: UDP 987, TCP 997.
+        failed = await self.hass.async_add_executor_job(helper.port_bind)
+        if failed is not None:
+            reason = 'port_{}_bind_error'.format(failed)
+            return self.async_abort(reason=reason)
+        return await self.async_step_creds(user_input)
+
+    async def async_step_creds(self, user_input=None):
+        """Return PS4 credentials from 2nd Screen App."""
+        from pyps4_homeassistant import Helper
+        helper = Helper()
```

Comment:
Since we're using this helper instance in many methods, we could save it as an instance attribute on the flow handler. Then we wouldn't need to import and instantiate it multiple times.

Reply:
Ok. I was trying to follow the imports must be in functions where it is used rule.

Reply:
If we import the helper class in init and save the helper instance as an attribute, we're following the rule.